Monsters hunting persistence/src/main/java/cz/fi/muni/pa165/dao/UserDaoImpl.java

```diff
@@ -1,35 +1,70 @@
 package cz.fi.muni.pa165.dao;
 
 import cz.fi.muni.pa165.entity.User;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
 
 import javax.inject.Named;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.PersistenceContext;
+import javax.persistence.NoResultException;
 import java.util.List;
 
+/**
+ * Implementation of UserDao interface
+ * @author Balcirak Peter <peter.balcirak@gmail.com>
+ */
 @Named
 public class UserDaoImpl implements UserDao {
+
+	@PersistenceContext
+	private EntityManager em;
+
 	@Override
-	public void create(User user) {
-		throw new NotImplementedException();
+	public void create(User user) throws IllegalArgumentException {
+		if(user == null){
+			throw new IllegalArgumentException("Parameter User cannot be null!");
+		}
+		em.persist(user);
 	}
 
 	@Override
-	public void delete(User user) {
-		throw new NotImplementedException();
+	public void delete(User user) throws IllegalArgumentException {
+		if(user == null){
+			throw new IllegalArgumentException("Parameter User cannot be null!");
+		}
+		em.remove(user);
 	}
 
 	@Override
-	public User findById(Long id) {
-		throw new NotImplementedException();
+	public User findById(Long id) throws IllegalArgumentException {
+		if(id == null){
+			throw new IllegalArgumentException("Parameter id cannot be null!");
+		}
+		return em.find(User.class, id);
 	}
-	
+
 	@Override
-	public User findByEmail(String email) {
-		throw new NotImplementedException();
+	public User findByEmail(String email) throws IllegalArgumentException {
+		if(email == null){
+			throw new IllegalArgumentException("Parameter email cannot be null!");
+		}
+		if(email.isEmpty()){
+			throw new IllegalArgumentException("Parameter email cannot be empty!");
+		}
+		try {
+			return em
+					.createQuery("select u from User u where email = :email",
+							User.class).setParameter("email", email)
+					.getSingleResult();
+		} catch (NoResultException nrf) {
+			return null;
+		}
 	}
 
 	@Override
 	public List<User> getAll() {
-		throw new NotImplementedException();
+		TypedQuery<User> query = em.createQuery("SELECT u FROM User u",
```

Comment:
The TypedQuery reference here is unnecessary. Call the `getResultingList()` right after the `createQuery` method.